### PR TITLE
test: api_portbindings_test: add new checkpoints

### PR DESCRIPTION
**Pull request by chuanchang**
Add new checkpoints into portbindings testing.

Signed-off-by: Alex Jia <chuanchang.jia@gmail.com>

### Ⅰ. Describe what this PR did
As summary.

### Ⅳ. Describe how to verify it
go test

### CODE REVIEW

File: apis/opts/portbindings_test.go [hunk 1]
```diff
@@ -12,14 +13,16 @@ func TestParsePortBinding(t *testing.T) {
...
-		args    args
-		want    types.PortMap
-		wantErr bool
+		name      string
+		args      args
+		want      types.PortMap
+		wantErr   bool
+		expectErr error
 	}{
 		// TODO: Add test cases.
```
Comment (fuweid) [inline apis/opts/portbindings_test.go]: could we remove this line?

File: apis/opts/portbindings_test.go [hunk 2]
```diff
@@ -40,15 +43,17 @@ func TestVerifyPortBinding(t *testing.T) {
...
 	tests := []struct {
-		name    string
-		args    args
-		wantErr bool
+		name      string
+		args      args
+		wantErr   bool
+		expectErr error
 	}{
 		// TODO: Add test cases.
```
Comment (fuweid) [inline apis/opts/portbindings_test.go]: could we remove this line?

File: apis/opts/portbindings_test.go (hunk 1 above)
Comment (chuanchang) [reply apis/opts/portbindings_test.go]: sure

File: apis/opts/portbindings_test.go (hunk 2 above)
Comment (chuanchang) [reply apis/opts/portbindings_test.go]: sure

File: apis/opts/portbindings_test.go [hunk 3]
```diff
@@ -40,15 +42,16 @@ func TestVerifyPortBinding(t *testing.T) {
...
-		name    string
-		args    args
-		wantErr bool
+		name      string
+		args      args
+		wantErr   bool
+		expectErr error
 	}{
-		// TODO: Add test cases.
 		{name: "testCase1", args: args{portBindings: map[string][]types.PortBinding{"21/ftp": {types.PortBinding{HostIP: "", HostPort: "21"}}}}, wantErr: false},
```
Comment (fuweid) [inline apis/opts/portbindings_test.go]: @chuanchang could we change the case name to be more specific? I don't think the `testCase1` is useful.

Comment (Ace-Tang) [reply apis/opts/portbindings_test.go]: testIpPort, testPortFTP, testPortTCP ?

Comment (chuanchang) [reply apis/opts/portbindings_test.go]: @fuweid the "testCase1" and "testCase2" originally exist in current code base, they are not added by me, in order to keep consistent with existent code, I added "testCase3" as new case.

```shell
-               // TODO: Add test cases.
                {name: "testCase1", args: args{ports: []string{"127.0.0.1:80:80"}}, want: map[string][]types.PortBinding{"80/tcp": {types.PortBinding{"127.0.0.1", "80"}}}, wantErr: false},
                {name: "testCase2", args: args{ports: []string{"26:22"}}, want: map[string][]types.PortBinding{"22/tcp": {types.PortBinding{"", "26"}}}, wantErr: false},
+               {name: "testCase3", args: args{ports: []string{"foo/bar"}}, expectErr: fmt.Errorf("Invalid containerPort: bar"), wantErr: true},
```
Of course, I can change case name together if it's necessary.

**Comment by chuanchang** [comment]
this part codes have been covered 100%, so close it.

**Comment by zhuangqh** [comment]
outdated change.  latest code has a full cover test. thus, closed @chuanchang

**Comment by chuanchang** [comment]
The b19cbf6 has fixed related issues, which are mentioned in this PR,  and cover 100% code path, so I tend to close this one again. BTW, I will give a extra PR to fix typo issue.

**Pull request closed without merging**